internal/manager/run.go

```diff
@@ -172,6 +172,7 @@ func Run(
 		featureGates,
 		routerFlavor,
 		c.UpdateStatus,
+		kongStartUpConfig.Version.IsKongGatewayEnterprise(),
```

Comment:
This will suffer for the same reasons router flavor suffers: it's being taken from the initial configuration and we assume it won't change.

While router flavor has implications such that the generated configs might break this will potentially try to configure enterprise features/entities (currently only licenses) even after a change from EE to OSS has been made (or the other way around: users won't get EE features until they restart KIC).

There is no easy fix that I can see for this as of now unless we make translators be aware of each Kong instance's config and create those dynamically. This would be a nice UX improvement but might require some non-insignificant effort.

cc: @mheap

Reply:
Heterogeneous Kong gateways!! This is a big feature which could bring up better UX and less downtime. Maybe also required by KGO.

Reply:
Linking https://github.com/Kong/kubernetes-ingress-controller/issues/3590 for reference.